storage/innobase/row/row0merge.cc

```diff
@@ -5178,6 +5178,8 @@ dberr_t row_merge_bulk_t::write_to_tmp_file(ulint index_no)
                        buf->index->table->space->id))
     return DB_TEMP_FILE_WRITE_FAIL;
   MEM_UNDEFINED(&m_block[0], srv_sort_buf_size);
+  DBUG_EXECUTE_IF("write_to_tmp_file_fail",
+                  return DB_TEMP_FILE_WRITE_FAIL;);
```

Comment:
Because we’re skipping the `MEM_UNDEFINED()` if `row_merge_write()` fails, also the fault injection should be before the `MEM_UNDEFINED()`.